<launch>

	<arg name="namespace"		default="pepper01" />



	// Launch nodes in pepper01 namespace ----------------------------------------
	<group ns="$(arg namespace)">


		// Pepper01 launch in pepper01 namespace -------------------------------------
		<include file="$(find printeps_pepper_modules)/launch/pepper_setup2.launch">
			<arg name="nao_ip"              value="$(optenv NAO_IP 192.168.1.16)" />
			<arg name="namespace"           value="$(arg namespace)" />
		</include>

		<!-- Robot LRS Offset Broadcaster -->
		<node pkg="tf" 	name="base_footprint_to_base_scan" type="static_transform_publisher" args="0.18 0 0 0 0 0 /$(arg namespace)/base_footprint /$(arg namespace)/base_scan 100" />


		<!-- Laser Range launch  -->
		<node ns="lrs_front" pkg="urg_node" type="urg_node" name="lrs0">
			<param name="ip_address" value="192.168.1.17"/>
			<param name="frame_id" 	value="$(arg namespace)/base_scan"/>
		</node>






	  <!-- Joystick motion -->
	  <arg name="joy_dev" default="/dev/input/js0" />

	  <node pkg="joy" type="joy_node" name="joy_node">
	    <param name="dev" value="$(arg joy_dev)" />
	    <param name="autorepeat_rate" value="5" />
	  </node>


		<!-- Pepper Navigatoin service -->
  	<node pkg="printeps_pepper_modules" name="pepper_navigation" type="pepper_navigation.py" >
			<remap from="lrs_front/scan"		to="$(arg namespace)/lrs_front/scan" />
			<remap from="amcl_pose" 				to="$(arg namespace)/amcl_pose" />
			<remap from="pepper_navigation" to="$(arg namespace)/pepper_navigation" />
		</node>



  	<!-- Display -->
  	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find printeps_pepper_modules)/rviz/$(arg namespace)_motion2.rviz" />

	</group>


	<!-- provide map -->
	<node name="map_server" pkg="map_server" type="map_server"
			args="$(find printeps_pepper_modules)/map/map.yaml" respawn="false" >
			<param name="use_map_topic"   value="true"/>
			<param name="frame_id" 				value="map" />
	</node>

	<!-- localizaion -->
	<node pkg="amcl" type="amcl" name="amcl" output="screen">
			<remap from="scan" to="$(arg namespace)/lrs_front/scan" />
			<param name="odom_frame_id"		value="$(arg namespace)/odom"/>
			<param name="base_frame_id"		value="$(arg namespace)/base_footprint"/>
			<param name="global_frame_id"	value="map"/>
			<rosparam file="$(find printeps_pepper_modules)/config/amcl.yaml" command="load" />
	</node>






	<node pkg="printeps_pepper_modules" name="teleop_twist_joy" type="teleop_twist_joy.py" >
		<remap from="joy" 			to="$(arg namespace)/joy" />
		<!--remap from="cmd_vel"		to="$(arg namespace)/cmd_vel" /-->
		<param name="max_speed"	value="0.25" />
		<param name="max_turn"	value="0.40" />
	</node>

</launch>
